Modules/rat_rescue.py

```diff
@@ -93,6 +93,49 @@ def __init__(self, case_id: UUID, client: str, system: str, irc_nickname: str,
         self._board_index = board_index
         self._lang_id = lang_id
 
+    def __eq__(self, other) -> bool:
+        """
+        Verify `other` is equal to the elf.
+
+        Args:
+            other (Rescue): Rescue to compare against
+
+        Returns:
+            bool: is equivalent
+        """
+        if not isinstance(other, Rescue):
+            # instance type check
+            raise TypeError(f"other was of type {type(other)} expected instance of Rescue")
+        else:
+            # check equality
+            try:
+                assert self.case_id == other.case_id
+                assert self.board_index == other.board_index
+                assert self.client == other.client
+                assert self.rats == other.rats
+                assert self.platform == other.platform
+                assert self.first_limpet == other.first_limpet
+                assert self.created_at == other.created_at
+                assert self.updated_at == other.updated_at
+                assert self.system == other.system
+                assert self.unidentified_rats == other.unidentified_rats
+                assert self.active == other.active
+                assert self.code_red == other.code_red
+                assert self.successful == other.successful
+                assert self.title == other.title
+                assert self.first_limpet == other.first_limpet
+                assert self.mark_for_deletion == other.mark_for_deletion
+                assert self._lang_id == other._lang_id  # TODO: implement property
+                assert self.rats == other.rats
+                assert self._irc_nick == other._irc_nick  # TODO: implement property
+            except AssertionError as ex:
+                raise ex
+                # if at least one of the above statements is false.
+                return False
```

Comment:
1. the `return False` is unreachable
2. this isn't a great way of doing things. have a look at the `all` built-in function
3. perhaps checking whether two rescue objects refer to the same rescue would be more useful than checking if all their attributes match. What I'm thinking of is to, when an update comes and the rescue is (or has become) open, check if that rescue is on our board and if so, modify it.

Reply:
1. i spotted it after i pushed and already fixed that.
2. i agree this probably isn't the best way of doing it, i was after something that worked. 
3. you asked for `__eq__`, which is to check if two object instances are equal. That is different from asking if two objects are the same. If you want to check if a newly created Rescue is a duplicate of an existing, perhaps a check against `search_by_name` may work here instead.

Reply:
Il look into a more clean solution for 2.
i can probably loop over attributes or something

Reply:
ok, so in this instance i think using a non-standard behavior for __contains__ may be best.
When an update comes in / rescue object is to be created it will check to see if there is *an* rescue on the board that already matches whats about to be created.

The check would have to verify guid, client name, and/or platform?